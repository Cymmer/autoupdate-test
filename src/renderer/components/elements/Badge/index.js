import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Span from '../Text/Span';
import Icon from '../Icon';

const Badge = ({ className, icon, id, text, colorName }) => (
  <div className={cn(className, styles[`Badge___${colorName}`])} id={id}>
    {icon && <Icon className={styles.Badge_icon} icon={icon} />}
    <Span
      className={cn({
        [styles.Badge_text___withIcon]: icon,
        [styles.Badge_text___noIcon]: !icon,
      })}
      id={id}
    >
      {text}
    </Span>
  </div>
);

Badge.defaultProps = {
  className: null,
  text: null,
  icon: null,
  id: null,
};

Badge.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string,
  id: PropTypes.string,
  colorName: PropTypes.oneOf([
    GLOBALS.COLOR_NAMES.GREEN,
    GLOBALS.COLOR_NAMES.BLUE,
    GLOBALS.COLOR_NAMES.RED,
    GLOBALS.COLOR_NAMES.YELLOW,
    GLOBALS.COLOR_NAMES.DARK_YELLOW,
    GLOBALS.COLOR_NAMES.DARK_GREEN,
    GLOBALS.COLOR_NAMES.DARK_BLUE,
    GLOBALS.COLOR_NAMES.GRAY,
  ]).isRequired,
  text: PropTypes.string,
};

export default Badge;
