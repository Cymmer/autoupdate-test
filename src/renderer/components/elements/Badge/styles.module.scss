@import '../../../scss/variables';
@import '../../../scss/mixins';
@import '../../../scss/themes';

@mixin base($bg-color, $color) {
  @include flex-center;
  background-color: $bg-color;
  color: $color;
  border-radius: 16px;
  padding: 6px 8px;
  margin: 4px;

  @include breakpoint(sm) {
    padding: 6px;
  }
}

.Badge {
  &___green {
    @include base(transparentize($green-300, 0.9), $green-300);
  }

  &___red {
    @include base(transparentize($red-300, 0.9), $red-300);
  }

  &___blue {
    @include base(transparentize($blue-300, 0.9), $blue-300);
  }

  &___yellow {
    @include base(transparentize($yellow-300, 0.9), $yellow-300);
  }

  &___darkYellow {
    @include base($yellow-300, $neutral-700);
  }

  &___darkGreen {
    @include base($green-300, $neutral-50);
  }

  &___darkBlue {
    @include base($blue-300, $neutral-50);
  }

  &___gray {
    @include flex-center;
    @include theme {
      color: t('tagColor');
      background-color: t('tagBackgroundColor');
    }
    border-radius: 16px;
    padding: 6px 8px;
    margin: 4px;

    @include breakpoint(sm) {
      padding: 6px;
    }
  }

  &_text {
    &___withIcon {
      font-size: 0.75rem;
      font-weight: 600;

      @include breakpoint(sm) {
        display: none;
      }
    }

    &___noIcon {
      font-size: 0.75rem;
      font-weight: 600;

      @include breakpoint(sm) {
        font-size: 0.625rem;
      }
    }
  }

  &_icon {
    margin: 0 4px 0 0;

    @include breakpoint(sm) {
      margin: 0;
    }
  }
}
