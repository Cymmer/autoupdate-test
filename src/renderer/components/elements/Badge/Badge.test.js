import React from 'react';
import { shallow } from 'enzyme';
import Badge from './index';

describe('RENDERING - Badge', () => {
  it('should render with default props', () => {
    const component = shallow(<Badge />);
    expect(component).toMatchSnapshot();
  });

  it('should render with icon', () => {
    const component = shallow(<Badge icon="icon" />);
    expect(component).toMatchSnapshot();
  });
});
