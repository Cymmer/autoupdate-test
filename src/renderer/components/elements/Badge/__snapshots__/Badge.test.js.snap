// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Badge should render with default props 1`] = `
<div
  className="Badge___undefined"
  id={null}
>
  <Span
    className="Badge_text___noIcon"
    colorClass={null}
    id={null}
    type={null}
  />
</div>
`;

exports[`RENDERING - Badge should render with icon 1`] = `
<div
  className="Badge___undefined"
  id={null}
>
  <Icon
    className="Badge_icon"
    icon="icon"
    id={null}
    isPulsating={false}
    style={null}
  />
  <Span
    className="Badge_text___withIcon"
    colorClass={null}
    id={null}
    type={null}
  />
</div>
`;
